<launch>

  <arg name="rviz" default="false" />

  <group if="$(arg rviz)">
    <include file="$(find path_smoothing_ros)/launch/rviz.launch" />
  </group>

  <!-- marker to simple path -->
  <node name="path_from_marker" pkg="path_smoothing_ros" type="path_from_markers" output="screen">
    <!-- sub -->
    <remap from="/marker" to="/marker"/>
    <remap from="/generate_path" to="/generate_path"/>

    <!-- pub -->
    <remap from="/initial_path" to="/initial_path"/>

  </node>
  <!-- simple path to smooth path -->
  <node name="path_smoother" pkg="path_smoothing_ros" type="path_smoother" output="screen">
    <param name="points_per_unit" value="3"/>
    <param name="skip_points" value="0"/>
    <param name="use_end_conditions" value="true"/>
    <param name="use_middle_conditions" value="true"/>
    <!-- sub -->
    <remap from="/initial_path" to="/initial_path"/>
    <remap from="/initial_pose" to="/initial_pose"/>
    <remap from="/final_pose" to="/final_pose"/>
    <!-- pub -->   
    <remap from="/smoothed_path" to="/smoothed_path"/>
  </node>

  <node name="marker_demo" pkg="path_smoothing_ros" type="marker_demo" output="screen"/>


</launch>
